### ISSUE TITLE
Definition of the semantics of conversion scripts in MSL is missing

### ISSUE THREAD

**Issue by modelica-trac-importer** [tbeu/ModelicaSpecification#394]
**Reported by fcasella on 28 Jun 2013 08:54 UTC**
The development of the MSL relies on conversion scripts to handle name changes, relocation of models among packages, obsolete or deprecated models, etc. (see, e.g., ticket:1185#comment:1)

Unfortunately, the semantics of such conversion scripts is still undefined, both in Modelica 3.2rev2 (sect. 17.7.2) and in Modelica 3.3 (sect. 18.8.2).

If we plan on using conversion scripts for the development of the MSL, then we obviously need to standardize the semantics of such scripts as well.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1186

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 2 Jul 2013 10:44 UTC**
The operations used in conversion scripts in MSL are now documented in the conversion scripts themselves.

However, standarizing conversion scripts in MSL should be a task for the future version of MSL that has a conversion script (3.4? 4.0?).

The reason is that every time there has been conversion scripts it was necessary to have special solution for the special cases (e.g. vectorization in MLS 1->2, balancing in MSL 2->3); and those special cases were most of the work (but are no longer as relevant) - and it might be that we do not need to standarize all of these old conversions.

It also seems very likely something similar will be needed the next time as well.

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 18 Jul 2013 12:26 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 23 Oct 2013 13:36 UTC**
Instead of just pushing lots of tickets beween meeting, I propose to move it to latest milestone - feel try to change it back if there is more information.

As indicated above it will for MSL be needed for the next MSL that contains conversion scripts - so have that as milestone.

It would be good to have standarized conversion scrips before that (for user libraries), but then we would need a description of the desired functionality for the user libraries (it could be a sub-set of the functionality in the existing scripts; the question is what subset).

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 1 Aug 2015 21:36 UTC**
Milestone renamed

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 1 Aug 2015 21:39 UTC**
Milestone renamed

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 9 Mar 2016 08:20 UTC**
Duplicate of #1622 (or at least resolved by that MCP).
Note that making conversion scripts for new MSL version isn't covered by these tickets, but that can probably be done in parallel with the other changes in MSL.
